pkg/oc/cli/admin/release/extract_tools.go

```diff
@@ -0,0 +1,412 @@
+package release
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	imagereference "github.com/openshift/origin/pkg/image/apis/image/reference"
+	"github.com/openshift/origin/pkg/oc/cli/image/extract"
+)
+
+// extractTarget describes how a file in the release image can be extracted to disk.
+type extractTarget struct {
+	OS      string
+	Command string
+
+	TargetName string
+
+	InjectReleaseImage bool
+
+	ArchiveFormat string
+	AsArchive     bool
+	AsZip         bool
+
+	Mapping extract.Mapping
+}
+
+// extractTools extracts all referenced commands as archives in the target dir.
+func (o *ExtractOptions) extractTools() error {
+	return o.extractCommand("")
+}
+
+// extractTools extracts specific commands out of images referenced by the release image.
+// TODO: in the future the metadata this command contains might be loaded from the release
+//   image, but we must maintain compatibility with older payloads if so
+func (o *ExtractOptions) extractCommand(command string) error {
+	// Available targets is treated as a GA API and may not be changed without backwards
+	// compatibility of at least N-2 releases.
+	availableTargets := []extractTarget{
+		{
+			OS:      "darwin",
+			Command: "oc",
+			Mapping: extract.Mapping{Image: "cli-artifacts", From: "usr/share/openshift/mac/oc.exe"},
```

Comment:
I think Mac binaries don't end with `.exe` but I might be wrong not owning Mac :wink:

Reply:
Outdated.